engine/collection/test/cluster_switchover_test.go

```diff
@@ -444,8 +448,10 @@ func RunTestCase(tc *ClusterSwitchoverTestCase) {
 	// build halfway through the grace period for the epoch 1 cluster
 	tc.builder.AddBlocksWithSeals(flow.DefaultTransactionExpiry/2, 1)
 
-	epoch1 := tc.State().Final().Epochs().Previous()
-	epoch2 := tc.State().Final().Epochs().Current()
+	epoch1, err := tc.State().Final().Epochs().Previous()
+	require.NoError(tc.T(), err)
+	epoch2, err := tc.State().Final().Epochs().Current()
```

Comment:
Please only query the Final State's snapshot once (the latest finalized block may change between queries)
```suggestion
	finalSnap := tc.State().Final()
	epoch1, err := finalSnap.Epochs().Previous()
	require.NoError(tc.T(), err)
	epoch2, err := finalSnap.Epochs().Current()
```